packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -41,23 +42,31 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
   useHour12 = false,
   timeRange,
   strings = getDefaultStrings(useHour12, showSeconds),
-  defaultValue,
-  onChange,
+  currentDate,
+  onTimeChange,
   onFormatDate,
   onValidateUserInput,
+  placeholder = strings.defaultTimePickerPlaceholder,
   ...rest
 }: ITimePickerProps) => {
   const [userText, setUserText] = React.useState<string>('');
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>();
   const [errorMessage, setErrorMessage] = React.useState<string>('');
 
   const optionsCount = getDropdownOptionsCount(increments, timeRange);
 
-  const initialValue = useConst(defaultValue || new Date());
-  const baseDate: Date = React.useMemo(() => generateBaseDate(increments, timeRange, initialValue), [
-    increments,
-    timeRange,
-    initialValue,
-  ]);
+  const baseDate = React.useMemo(() => {
+    const initialDate = currentDate || new Date();
+    return generateBaseDate(increments, timeRange, initialDate);
+  }, [increments, timeRange, currentDate]);
+
+  React.useEffect(() => {
+    if (onTimeChange && !errorMessage && userText) {
+      const currentChosenTime = userText;
```

Comment:
nit: you don't need `currentChosenTime` since you just pass it to `getDateFromTimeSelection` on the next line. You can just pass `userText` directly to that function (unless I'm missing something?)

Reply:
This code snippet is no longer relevant and has been removed.